a3fe/tests/test_engine_configuration.py

```diff
@@ -0,0 +1,81 @@
+"""Unit and regression tests for the SomdConfig class."""
+
+from tempfile import TemporaryDirectory
+
+from a3fe import SomdConfig
+
+import os
+
+
+def test_create_config():
+    """Test that the config can be created."""
+    config = SomdConfig()
+    assert isinstance(config, SomdConfig)
+
+
+def test_config_pickle_and_load():
+    """Test that the config can be pickled and loaded."""
+    with TemporaryDirectory() as dirname:
+        config = SomdConfig()
+        config.dump(dirname)
+        config2 = SomdConfig.load(dirname)
+        assert config == config2
+
+
+def test_get_somd_config():
+    """
+    Test that the SOMD configuration file is generated correctly
+    and that the file is written correctly.
+    """
+    # Tmpdir to store the config
+    with TemporaryDirectory() as dirname:
+        config = SomdConfig(
```

Comment:
Once more validation is added, check that the more complex validation works as expected.